app/models/order_item.rb

```diff
@@ -0,0 +1,25 @@
+class OrderItem < ApplicationRecord
+  belongs_to :book
+  belongs_to :order
+
+  validates :quantity, presence: true, numericality: { only_integer: true, greater_than: 0 }
+  validate :book_present
+  validate :order_present
+
+  before_save :finalize
+
+
+private
+  def book_present
+    errors.add(:book, 'is not valid or is not active.') if book.nil?
```

Comment:
Instead of this it's possible to use `validates :book, presence: { message:  'is not valid or is not active.' }` or similar